runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -234,7 +234,20 @@ void __catalyst__rt__print_tensor(OpaqueMemRefT *c_memref, bool printDescriptor)
 
 void __catalyst__rt__fail_cstr(const char *cstr) { RT_FAIL(cstr); }
 
-void __catalyst__rt__initialize() { CTX = std::make_unique<ExecutionContext>(); }
+void __catalyst__rt__initialize(char *seed /* default value = nullptr in declaration */)
+{
+    if (seed == nullptr){
+        // Preserve legacy usage where __catalyst__rt__initialize is called without any arguments
+        CTX = std::make_unique<ExecutionContext>();
+    }
+    else if (strcmp(seed, "__catalyst__unseeded__qjit__run__") == 0) {
+        CTX = std::make_unique<ExecutionContext>();
```

Comment:
I think this block and the one above can be merged together, but I'm not sure how to produce a nullptr global in the mlir conversion. If we just set the default stringref to empty string the produced entity is `zeroinitializer`, not nullptr.